src/features/ci-cd/models/release-additions.ts

```diff
@@ -0,0 +1,159 @@
+import type {
+	AbstractSqlModel,
+	CastNode,
+	ConcatenateNode,
+	ExistsNode,
+} from '@balena/abstract-sql-compiler';
+
+export const addToModel = (abstractSql: AbstractSqlModel) => {
+	const isFinal: ExistsNode = [
+		'Exists',
+		['ReferencedField', 'release', 'revision'],
+	];
+
+	abstractSql.tables['release'].fields.push({
+		fieldName: 'is final',
+		dataType: 'Boolean',
+		required: true,
+		computed: isFinal,
+	});
+
+	const semverField: ConcatenateNode = [
+		'Concatenate',
+		['ReferencedField', 'release', 'semver major'],
+		['EmbeddedText', '.'],
+		['ReferencedField', 'release', 'semver minor'],
+		['EmbeddedText', '.'],
+		['ReferencedField', 'release', 'semver patch'],
+	];
+
+	abstractSql.tables['release'].fields.push({
+		fieldName: 'semver',
+		dataType: 'Short Text',
+		required: true,
+		computed: semverField,
+	});
+
+	const createdAtTimestamp: CastNode = [
+		'Cast',
+		[
+			'Floor',
+			[
+				'Multiply',
+				['Totalseconds', ['ReferencedField', 'release', 'created at']],
+				['Number', 1000],
```

Comment:
Any particular reason for this timestamp conversion?

Reply:
The DB's `EXTRACT(EPOCH FROM A_TIMESTAMP_HERE)` gives something like `1,627,388,902.838732` which uses a different accuracy & format than the nodejs Date's numeric representation which gives something like `1,627,389,065,266`.
The requirement for the version field of draft releases was to have that `-${timestamp}` prerelease part. Now the floor & * 1000 is strictly speaking not required, but thought it would be better if that was consistent with the js numeric representation of the `created_at` field.

Reply:
Aha, good to know about the mismatch in timestamp nature between DB and JS. TIL.